react/src/components/organisms/Header/index.js

```diff
@@ -1,35 +1,119 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-
+import is from 'is';
+import rfdc from 'rfdc';
 import UtilityNav from '../UtilityNav';
 import MainNav from '../../molecules/MainNav';
 import HeaderSearch from '../../molecules/HeaderSearch';
 import SiteLogo from '../../atoms/media/SiteLogo';
 import logo from '../../../assets/images/stateseal.png';
 import './styles.css';
 
+const clone = rfdc();
 class Header extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      utilNavOpen: false
+      utilNavOpen: false,
+      shouldNavigateTop: false,
+      shouldNavigateBottom: true
     };
+    this.searchInputTop = React.createRef();
+    this.searchInputBottom = React.createRef();
+    this.buttonRefTop = React.createRef();
+    this.buttonRefBottom = React.createRef();
   }
-  menuButtonClicked() {
-    const bodyClass = document.querySelector('body').classList;
-    bodyClass.toggle('show-menu');
-    if (bodyClass.value.length > 0) {
-      this.setState({ utilNavOpen: false });
+  menuButtonClicked = (afterButtonSearch = false) => {
+    // eslint-disable-next-line no-undef
+    const body = document.querySelector('body');
+    let bodyClass;
+    if (body.hasAttribute('class')) {
+      bodyClass = body.getAttribute('class');
+      if (bodyClass) {
+        if (bodyClass.indexOf('show-menu') !== -1) {
+          bodyClass = bodyClass.split(' ').filter((val) => val !== 'show-menu').join(' ');
+        } else {
+          bodyClass += ' show-menu';
+        }
+        body.setAttribute('class', bodyClass);
+      } else {
+        bodyClass = 'show-menu';
+        body.setAttribute('class', 'show-menu');
+      }
+    } else {
+      bodyClass = 'show-menu';
+      body.setAttribute('class', 'show-menu');
+    }
+    const newState = (bodyClass) ? { utilNavOpen: false } : { utilNavOpen: true };
+    if (afterButtonSearch) {
+      this.setState(newState, this.afterButtonSearch);
     } else {
-      this.setState({ utilNavOpen: true });
+      this.setState(newState);
+    }
+  };
+  // Puts focus on the mobile search input after the menu is opened via the search button.
+  afterButtonSearch = () => {
+    document.getElementById('nav-search').focus();
+    if (is.fn(this.props.headerSearch.buttonSearch.onClick)) {
+      this.props.headerSearch.buttonSearch.onClick();
+    }
+  };
+  // On click action used for both top and bottom buttons.
+  defaultButtonSearchOnClick = (e) => {
+    if (e && e.currentTarget === this.buttonRefTop.current) {
+      if (this.state.shouldNavigateTop) {
+        const query = this.searchInputTop.current.value;
+        if (query.length > 0) {
+          window.location.href = `https://search.mass.gov/?q=${query}`;
```

Comment:
i think this should be the default behavior but you should be able to override to go to a different search page -- e.g the budget site search page.